lepton/openstack_dns.go

```diff
@@ -0,0 +1,106 @@
+package lepton
+
+import (
+	"os"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/dns/v2/recordsets"
+	"github.com/gophercloud/gophercloud/openstack/dns/v2/zones"
+)
+
+// FindOrCreateZoneIDByName searches for a DNS zone with the name passed by argument and if it doesn't exist it creates one
+func (o *OpenStack) FindOrCreateZoneIDByName(config *Config, dnsName string) (string, error) {
+	dnsClient, err := o.getDNSClient()
+	if err != nil {
+		return "", err
+	}
+
+	var zoneID string
+
+	opts := zones.ListOpts{
+		Name: dnsName + ".",
+	}
+	allPages, err := zones.List(dnsClient, opts).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	allZones, err := zones.ExtractZones(allPages)
+	if err != nil {
+		return "", err
+	}
+
+	if len(allZones) == 0 {
+		createOpts := zones.CreateOpts{
+			Name:  dnsName + ".",
+			Email: "martinhoferreira10@gmail.com",
```

Comment:
? is this required or ?

Reply:
I added my personal email by accident. Without email parameter, it isn't possible to create a zone in openstack.

Reply:
that seems weird? if you need something you can just use the example.com from http://www.faqs.org/rfcs/rfc2606.html

Reply:
I added a default email with format `admin@<domain name`. I guess it makes sense the user to add a command flag that sets his email.